### DISCUSSION TITLE
Debounce a controlled input

### DISCUSSION THREAD

**Discussion by RiddlePartyGames** [react-hook-form/react-hook-form#6946] [Q&A]
We've migrated to using react hook forms and so far it has been great. However now we are experiencing an issue where our form is becoming heavy.  Is it possible call a debounce on an onChange method within a controlled component so that our components don't render after every key stroke? 

```
<Controller
  control={control}
  name="test"
  render={({
    field: { onChange, onBlur, value, name, ref },
    fieldState: { invalid, isTouched, isDirty, error },
    formState,
  }) => (
    <Checkbox
      onBlur={onBlur}
      onChange={onChange}
      checked={value}
      inputRef={ref}
    />
  )}
/>
```

**Comment by bluebill1049** [comment]
can you define the heavy part? Controller's rerender should be isolated, so debounce is probably not going to help out your existing perf issue.
